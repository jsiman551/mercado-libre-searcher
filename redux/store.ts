import { configureStore } from '@reduxjs/toolkit'
import searchDataReducer from './slices/search-data-slice'
import searchInputReducer from './slices/search-input-slice'
import sortOptionReducer from './slices/sort-option-slice'
import sidebarFlagReducer from './slices/sidebar-flag-slice'
import priceRangeReducer from './slices/price-range-slice'

export const store = configureStore({
  reducer: {
    searchData: searchDataReducer,
    searchInput: searchInputReducer,
    sortOption: sortOptionReducer,
    mobileSidebarFlag: sidebarFlagReducer,
    priceRange: priceRangeReducer,
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
