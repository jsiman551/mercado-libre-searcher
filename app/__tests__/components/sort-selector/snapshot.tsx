import { render } from '@testing-library/react';
import SortSelector from '@/components/sort-selector';
import { store } from '@/redux/store';
import { Provider } from 'react-redux';

it('renders sort opctions selector unchanged', () => {
  const { container } = render(
    <Provider store={store}>
      <SortSelector />
    </Provider>
  );
    expect(container).toMatchSnapshot();
});
