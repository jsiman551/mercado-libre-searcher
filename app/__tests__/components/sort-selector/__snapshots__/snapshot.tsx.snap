// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders sort opctions selector unchanged 1`] = `
<div>
  <div
    class="sc-d2de8d94-0 fQlvBD"
    data-testid="sortContainer"
  >
    <button
      class="sc-d2de8d94-7 dndIoj"
    >
      <img
        alt="open sidebar"
        data-nimg="1"
        decoding="async"
        height="20"
        loading="lazy"
        src="/icons/burger-icon.svg"
        style="color: transparent;"
        width="20"
      />
    </button>
    <div
      class="sc-d2de8d94-1 FoZaU"
    >
      <div
        class="sc-d2de8d94-2 fURfgz"
      >
        <p>
          Ordenar por
        </p>
      </div>
      <div
        class="sc-d2de8d94-3 fJdPS"
      >
        <p>
          Selecciona
        </p>
        <img
          alt="selector"
          class="sc-d2de8d94-4 heBvGm"
          height="16"
          src="/icons/arrow-selector-icon.svg"
          width="16"
        />
      </div>
    </div>
  </div>
</div>
`;
